dbms/src/Common/Config/ConfigReloader.cpp

```diff
@@ -73,29 +73,68 @@ void ConfigReloader::run()
     }
 }
 
+bool ConfigReloader::tlsCertUpdated(Poco::AutoPtr<Poco::Util::AbstractConfiguration> config, bool init)
+{
+    try
+    {
+        LOG_DEBUG(log, "check tls cert file updated");
+        if (config->has("security") || config->has("security.ca_path") || config->has("security.cert_path") || config->has("security.key_path"))
+        {
+            if (init)
+            {
+                cert_files.addIfExists(config->getString("security.ca_path"));
+                cert_files.addIfExists(config->getString("security.cert_path"));
+                cert_files.addIfExists(config->getString("security.key_path"));
+            }
+            else
+            {
+                FilesChangesTracker new_files;
+                new_files.addIfExists(config->getString("security.ca_path"));
+                new_files.addIfExists(config->getString("security.cert_path"));
+                new_files.addIfExists(config->getString("security.key_path"));
+                bool updated = cert_files.isDifferOrNewerThan(new_files);
+                if (updated)
+                {
+                    cert_files = std::move(new_files);
+                }
+                return updated;
+            }
+        }
+        return false;
+    }
+    catch (...)
+    {
+        tryLogCurrentException(log, "Error loading config from `" + path + "'");
+        return false;
+    }
+}
+
 void ConfigReloader::reloadIfNewer(bool force, bool throw_on_error)
 {
     std::lock_guard lock(reload_mutex);
 
     FilesChangesTracker new_files = getNewFileList();
-    if (force || new_files.isDifferOrNewerThan(files))
+
+    ConfigProcessor config_processor(path);
```

Comment:
I think we don't need to load the config everytime, if `config` itself is not changed, the tls file path can be found in `cert_files`